File: _Basic/windows-forms-introduction-with-visual-basic/2-windows-forms-introduction-with-visual-basic-m2-forms2-exercise-files/CS/UserControls/UserControls/PersonControl.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace UserControls
{
    public partial class PersonControl : UserControl
    {
        public PersonControl()
        {
            InitializeComponent();
        }

        private Person _person;

        public Person Person
        {
            get
            {
                _person.FirstName = FirstTextBox.Text;
                _person.LastName = LastNameTextBox.Text;
                _person.Age = Convert.ToInt32(AgeTextBox.Text);
                return _person;
            }
            set
            {
                FirstTextBox.Text = value.FirstName;
                LastNameTextBox.Text = value.LastName;
                AgeTextBox.Text = value.Age.ToString();
                _person = value;
            }
        }
    }
}

File: _Basic/windows-forms-introduction-with-visual-basic/2-windows-forms-introduction-with-visual-basic-m2-forms2-exercise-files/VB/UserControls/UserControls/PersonControl.vb
Public Class PersonControl
    Private _person As Person

    Public Property Person() As Person
        Get
            _person.FirstName = FirstTextBox.Text
            _person.LastName = LastNameTextBox.Text
            _person.Age = CInt(AgeTextBox.Text)
            Return _person
        End Get
        Set(ByVal value As Person)
            FirstTextBox.Text = value.FirstName
            LastNameTextBox.Text = value.LastName
            AgeTextBox.Text = value.Age
            _person = value
        End Set
    End Property

End Class
